src/test/java/com/example/rets_api/repository/PropertyRepositoryJPATest.java

```diff
@@ -138,6 +141,30 @@ public void should_update_a_Property() {
         comparePropertyPatchDTOBasicFields(PropertyConverter.propertyEntityToPropertyPatchDTO.convert(responseEntity), PropertyConverter.propertyEntityToPropertyPatchDTO.convert(propertyToPatch));
     }
 
+    @Test
+    public void should_update_a_Property_updateViewData() {
+        //Creating
+        ViewDataDTO viewDataDTOToPatch = createViewDataDTO();
+        ViewDataEntity viewDataEntityToPatch = ViewDataConverter.viewDataDTOToViewDataEntity.convert(viewDataDTOToPatch);
+        PropertyEntity propertyToPatch = EntityUtilsTest.createPropertyEntityWithDifferentBasicFields();
+        propertyToPatch.setViewData(viewDataEntityToPatch);
+        PropertyEntity propertyEntity = EntityUtilsTest.createPropertyEntityWithBasicFields();
+
+        //Saving
+        propertyEntity = propertyRepository.saveAndFlush(propertyEntity);
+
+        //Updating
+        PropertyEntity propertyToSave = updatePropertyValues(propertyEntity, propertyToPatch);
+
```

Comment:
here updatePropertyValues method only updates the basic fields it doesn't update the viewdata

Reply:
that is why I add the viewData to it in following lines

Reply:
done :)